dev-java/jna/files/jna-5.13.0-JNALoadTest.patch

```diff
@@ -0,0 +1,59 @@
+1) testLoadAndUnloadFromJar(com.sun.jna.JNALoadTest)
+java.lang.UnsatisfiedLinkError: Native library (com/sun/jna/linux-x86-64/libjnidispatch.so) not found in resource path ([file:/var/tmp/portage/dev-java/jna-5.13.0/work/jna-5.13.0/build/jna.jar])
+	at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:1059)
+	at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:1015)
+	at com.sun.jna.Native.<clinit>(Native.java:221)
+	at java.base/java.lang.Class.forName0(Native Method)
+	at java.base/java.lang.Class.forName(Class.java:467)
+	at com.sun.jna.JNALoadTest.testLoadAndUnloadFromJar(JNALoadTest.java:128)
+2) testAvoidResourcePathLoading(com.sun.jna.JNALoadTest)
+java.lang.Error: Expected JNA library at build/classes/com/sun/jna/linux-x86-64/libjnidispatch.so is missing
+	at com.sun.jna.JNALoadTest.assertLibraryExists(JNALoadTest.java:87)
+	at com.sun.jna.JNALoadTest$TestLoader.<init>(JNALoadTest.java:54)
+	at com.sun.jna.JNALoadTest.testAvoidResourcePathLoading(JNALoadTest.java:109)
+3) testLoadFromUnicodePath(com.sun.jna.JNALoadTest)
+java.lang.Error: JVM error: System.load() failed to load JNA native library from /var/tmp/portage/dev-java/jna-5.13.0/work/jna-5.13.0/build/native/libjnidispatch.so): java.lang.UnsatisfiedLinkError: Native library (com/sun/jna/linux-x86-64/libjnidispatch.so) not found in resource path ([file:/var/tmp/portage/dev-java/jna-5.13.0/work/jna-5.13.0/build/jna.jar])
+	at com.sun.jna.JNALoadTest.testLoadFromUnicodePath(JNALoadTest.java:266)
+4) testLoadAndUnloadFromResourcePath(com.sun.jna.JNALoadTest)
+java.lang.Error: Expected JNA library at build/classes/com/sun/jna/linux-x86-64/libjnidispatch.so is missing
+	at com.sun.jna.JNALoadTest.assertLibraryExists(JNALoadTest.java:87)
+	at com.sun.jna.JNALoadTest$TestLoader.<init>(JNALoadTest.java:54)
+	at com.sun.jna.JNALoadTest.testLoadAndUnloadFromResourcePath(JNALoadTest.java:184)
```

Comment:
what is exactly the problem with this? is the .so file just placed elsewhere or is it something different?

Reply:
This test class has 5 tests,one of them satisfied by [copying `jna.jar` to `build/`](https://github.com/gentoo/gentoo/pull/30966/files#diff-01b4b54a03cddc5e147d6f031b8c62831ad76920a24aac03c1a91f731471e089R157-R158). 
`libjnidispatch.so` is `build/native/libjnidispatch.so`. I've tried with adding it to `build/jna.jar` and with diverse modifications on `JNALoadTest.java` such like [this one](https://github.com/gentoo/gentoo/blob/master/dev-java/jna/files/jna-5.13.0-testpath.patch#L63C44-L64). Failure messages vary depending on what I try.
After all, I think this test class is not really useful for Gentoo since we know the native library exists and we finally install it using `java-pkg_doso`.
Would you agree to adding this test class to `JAVA_TEST_EXCLUDES`?

Reply:
Guess I got an idea what needs to be done.  
Gentoo installs `libjnidispatch.so` in `/usr/lib64/jna-4/`, so we could skip anything from this test class expecting it to sit in the jar file. Replicating everything what upstream does in [build.xml](https://github.com/java-native-access/jna/blob/5.13.0/build.xml#L1278-L1283) might get too complicated here.
Could you pls take it as a WIP for now?

Reply:
i agree that there is no point in testing loading the library from the jar.